app/controllers/participants_controller.rb

```diff
@@ -70,10 +70,19 @@ def need_more_times?
 
   # GET /participants/1
   # GET /participants/1.json
-  # def show
-  #   @participants = Participant.where(project_id: params[:project_id])
-  #   redirect_to projects_path
-  # end
+  def show
+    unless logged_in?
+      require_user
+      return
+    end
+
+    unless Participant.where(project_id: @project.id).ids.include? params[:id].to_i
```

Comment:
instead of `Participant.where(project_id: @project.id)` do `@project.participants`

Reply:
Thanks for adding!